File: Python/String-2/xyz_there.py
# Return True if the given string contains an appearance of "xyz" where the
# xyz is not directly preceeded by a period (.). So "xxyz" counts but
# "x.xyz" does not.

def xyz_there(str):
  # Check if xyz starts at index 0
  if(str[:3] == 'xyz'):
    return True
  
  # Check if xyz appears in the string and does not precede '.'
  for i in range(len(str) - 2):
    if(str[i:i + 3] == 'xyz' and str[i - 1] != '.'):
      return True
  
  return False

File: Java/String-2/xyzThere.java
/*
 * Return true if the given string contains an appearance of "xyz" where
 * the xyz is not directly preceeded by a period (.). So "xxyz" counts but
 * "x.xyz" does not.
 */

public boolean xyzThere(String str) {
  // Check to see if xyz appears and is not directly preceeded by a period(.)
  // First check if xyz appears at index 0, then check the rest of the string
  if(3 <= str.length() && str.substring(0, 3).equals("xyz"))
    return true;
  
  
  for(int i = 1; i < str.length() - 2; i++){
    if(str.substring(i, i + 3).equals("xyz") && str.charAt(i - 1) != '.')
      return true;
  }
  return false;
}
